### PR TITLE
Improve `testlib`

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * Updates `config`.
  * Extends `testlib` with utility class `Assertions`, adding methods for asserting throwing popular exceptions, and applies
     them to this project.
  * Introduces `TempDir` testing utility, which aims to replace `Files.createTempDir()` from Guava.
  * Improves layout of tests when they were touched by applying new testing utils.

### CODE REVIEW

**Review by dmdashenkov — APPROVED** [review]

File: tools/protoc-plugin/src/test/java/io/spine/tools/protoc/ExternalClassLoaderTest.java [hunk 1]
```diff
@@ -44,69 +46,78 @@
...
-            assertThrows(ClassInstantiationException.class,
-                         () -> newInstanceFor(NotMethodFactory.class));
+        void assertCannotInstantiate(Executable e) {
+            assertThrows(ClassInstantiationException.class, e);
         }
     }
 
-    @DisplayName("return a class instance by it's fully-qualified name")
     @Test
+    @DisplayName("return a class instance by it's fully-qualified name")
```
Comment (armiol) [inline tools/protoc-plugin/src/test/java/io/spine/tools/protoc/ExternalClassLoaderTest.java:120]: "its".

File: tools/protoc-plugin/src/test/java/io/spine/tools/protoc/iface/GenerateInterfacesTest.java [hunk 2]
```diff
@@ -45,34 +48,33 @@
...
     }
 
     @DisplayName("throw `IllegalArgumentException` if interface name is")
     @ParameterizedTest(name = "\"{0}\"")
     @ValueSource(strings = {"", "  "})
     void throwIllegalArgumentException(String interfaceName) {
-        assertThrows(IllegalArgumentException.class, () ->
-                newTask(newTaskConfig(interfaceName).build()));
+        assertIllegalArgument(() ->
+                                      newTask(newTaskConfig(interfaceName).build()));
```
Comment (armiol) [inline tools/protoc-plugin/src/test/java/io/spine/tools/protoc/iface/GenerateInterfacesTest.java:77]: It looks as it fits the previous line.

File: tools/protoc-plugin/src/test/java/io/spine/tools/protoc/iface/InterfaceGeneratorTest.java [hunk 3]
```diff
@@ -107,176 +108,186 @@ void notAcceptNulls() {
...
             }
         }
-    }
 
-    @Test
-    @DisplayName("generate insertion point contents for EveryIs in singe file")
-    void generateInsertionPointContentsForEveryIsInSingleFile() {
-        String filePath = "spine/tools/protoc/iface/every_is_in_one_file.proto";
+        @Test
+        @DisplayName("`EveryIs` in singe file")
```
Comment (armiol) [inline tools/protoc-plugin/src/test/java/io/spine/tools/protoc/iface/InterfaceGeneratorTest.java:168]: "single".

File: tools/protoc-plugin/src/test/java/io/spine/tools/protoc/iface/InterfaceGeneratorTest.java [hunk 4]
```diff
@@ -107,176 +108,186 @@ void notAcceptNulls() {
...
+        void generateInsertionPointContentsForEveryIsInSingleFile() {
```
Comment (armiol) [inline tools/protoc-plugin/src/test/java/io/spine/tools/protoc/iface/InterfaceGeneratorTest.java:169]: How is this test different from the next one called `void isInSingleFile`?

**Review by armiol — CHANGES_REQUESTED** [review]
@alexander-yevsyukov LGTM in general. Please see my comments.

File: tools/protoc-plugin/src/test/java/io/spine/tools/protoc/iface/InterfaceGeneratorTest.java (hunk 4 above)
Comment (alexander-yevsyukov) [reply tools/protoc-plugin/src/test/java/io/spine/tools/protoc/iface/InterfaceGeneratorTest.java:169]: This one is on `EveryIs`, while the next one is on `Is`.

**Review by armiol — APPROVED** [review]

**Pull request merged**